src/component/actions/SendMsg/SendMsgForm.tsx

```diff
@@ -122,23 +117,24 @@ export class SendMsgForm extends React.Component<SendMsgFormProps> {
     }
 
     public handleUpdateMessage(event: React.ChangeEvent<HTMLTextElement>): void {
-        this.props.updateSendMsgForm({ text: event.currentTarget.value });
+        const value = event.currentTarget.value;
+        this.props.updateSendMsgForm({
+            text: validate('Message', value, [validateRequired])
+        });
     }
 
-    public handleUpdateQuickReplies(quickReplies: string[]): void {
-        this.props.updateSendMsgForm({ quickReplies });
+    public handleUpdateQuickReplies(value: string[]): void {
+        this.props.updateSendMsgForm({
+            quickReplies: validate('Quick Replies', value, [validateMaxOfTen])
+        });
```

Comment:
These handlers here are the interesting bit, they call a shared validate method with a list of validators (which are also shared amongst all forms). These will return an object representing the actual value and also any errors if it is invalid.